### PR TITLE
NIAD-374: Empty patch objects are rejected

### PR DESCRIPTION
**Pull request by mattd-kainos**

### CODE REVIEW

File: src/intTest/resources/patient/amendment-empty-patch-object.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "nhsNumber": "9999999999",
+  "gpCode": "8880232,G82697",
+  "gpTradingPartnerCode": "TES5",
+  "healthcarePartyCode": "XX1",
+  "patches": [{}]
```
Comment (bartek-sarul) [inline src/intTest/resources/patient/amendment-empty-patch-object.json:6]: This will be `"patches" : [ ]` but no worries, I'll adjust this in my PR

**Review by bartek-sarul — APPROVED** [review]

File: src/intTest/resources/patient/amendment-empty-patch-object.json (hunk 1 above)
Comment (mattd-kainos) [reply src/intTest/resources/patient/amendment-empty-patch-object.json:6]: This test case is specifically for an array containing an empty object. There is another test for just an empty array.
